src/services/codefixes/fixMissingTypeAnnotationOnExports.ts

```diff
@@ -159,23 +188,248 @@ function addTypeAnnotationOnNode(node: Node, typeChecker: TypeChecker): Node | N
+            const typeNode = typeToTypeNode(type, node, typeChecker);
+            return changes.replaceNodeWithNodes(sourceFile, node, [
+                factory.createVariableStatement(/*modifiers*/ undefined,
+                    factory.createVariableDeclarationList(
+                        [factory.createVariableDeclaration(
+                            "__default", /*exclamationToken*/ undefined,
+                            typeNode, defaultExport.expression)],
+                        NodeFlags.Const)),
+                factory.updateExportAssignment(defaultExport, defaultExport?.modifiers, factory.createIdentifier("__default")),
+            ]);
+        }
+        break;
+    // Handling expression like heritage clauses e.g. class A extends mixin(B) ..
+    case SyntaxKind.ExpressionWithTypeArguments:
+        const expression = node as ExpressionWithTypeArguments;
+        const classDecl = expression.parent.parent as unknown as ClassDeclaration;
+        const heritageTypeNode = typeToTypeNode(
+            typeChecker.getTypeAtLocation(expression.expression),
+            expression.expression,
+            typeChecker);
+        const heritageVariableName = factory.createUniqueName(
+            classDecl.name? classDecl.name.text + "Base" : "Anonymous", GeneratedIdentifierFlags.Optimistic);
+        // e.g. const Point3DBase: typeof Point2D = mixin(Point2D);
+        const heritageVariable = factory.createVariableStatement(
+            /*modifiers*/ undefined,
+            factory.createVariableDeclarationList(
+                [factory.createVariableDeclaration(
+                    heritageVariableName,
+                    /*exclamationToken*/ undefined,
+                    heritageTypeNode,
+                    expression,
+                    )],
+                NodeFlags.Const,
+            )
+        );
+        changes.insertNodeAt(sourceFile, classDecl.getFullStart(), heritageVariable, { prefix: "\n" });
+        return changes.replaceNodeWithNodes(sourceFile, expression,
+            [factory.createExpressionWithTypeArguments(heritageVariableName, [])]);
+    case SyntaxKind.ObjectBindingPattern:
+    case SyntaxKind.ArrayBindingPattern:
+        return transformDestructuringPatterns(node as BindingPattern, sourceFile, typeChecker, changes);
     default:
         break;
     }
     throw new Error(`Cannot find a fix for the given node ${node.kind}`);
 }
 
+
+interface ExpressionReverseChain {
+    element?: BindingElement;
+    parent?: ExpressionReverseChain;
+    expression: SubExpression;
+}
+
+const enum ExpressionType {
+    TEXT = 0,
+    COMPUTED = 1,
+    ARRAY_ACCESS = 2,
+    IDENTIFIER = 3,
+}
+
+type SubExpression = {kind: ExpressionType.TEXT, text: string} |
+                     {kind: ExpressionType.COMPUTED, computed: Expression} |
+                     {kind: ExpressionType.ARRAY_ACCESS, arrayIndex: number} |
+                     {kind: ExpressionType.IDENTIFIER, identifier: Identifier} ;

```

Comment:
Formatting of unions would usually put the `|` in front:
Ex:

```ts
export type LateVisibilityPaintedStatement =
    | AnyImportSyntax
    | VariableStatement
    | ClassDeclaration
    | FunctionDeclaration
    | ModuleDeclaration
    | TypeAliasDeclaration
    | InterfaceDeclaration
    | EnumDeclaration;
    
```

Reply:
Done